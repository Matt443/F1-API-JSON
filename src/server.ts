import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}