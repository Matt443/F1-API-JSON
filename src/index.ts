import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
