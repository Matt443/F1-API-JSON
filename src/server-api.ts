import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
