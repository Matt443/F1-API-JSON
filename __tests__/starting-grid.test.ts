import { getStartingGrid } from "../src/server";

test("get race schedule", async () => {
    expect(await getStartingGrid(2024)).toMatchSnapshot();
});
