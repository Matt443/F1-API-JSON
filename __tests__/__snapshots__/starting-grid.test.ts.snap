// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get race schedule 1`] = `
[
  {
    "code": "VER",
    "name": "Max Verstappen",
    "number": 1,
    "position": 1,
    "team": "Red Bull Racing Honda RBPT",
    "time": "1:15.915",
  },
  {
    "code": "SAI",
    "name": "Carlos Sainz",
    "number": 55,
    "position": 2,
    "team": "Ferrari",
    "time": "1:16.185",
  },
  {
    "code": "NOR",
    "name": "Lando Norris",
    "number": 4,
    "position": 3,
    "team": "McLaren Mercedes",
    "time": "1:16.315",
  },
  {
    "code": "LEC",
    "name": "Charles Leclerc",
    "number": 16,
    "position": 4,
    "team": "Ferrari",
    "time": "1:16.435",
  },
  {
    "code": "PIA",
    "name": "Oscar Piastri",
    "number": 81,
    "position": 5,
    "team": "McLaren Mercedes",
    "time": "1:16.572",
  },
  {
    "code": "PER",
    "name": "Sergio Perez",
    "number": 11,
    "position": 6,
    "team": "Red Bull Racing Honda RBPT",
    "time": "1:16.274",
  },
  {
    "code": "RUS",
    "name": "George Russell",
    "number": 63,
    "position": 7,
    "team": "Mercedes",
    "time": "1:16.724",
  },
  {
    "code": "TSU",
    "name": "Yuki Tsunoda",
    "number": 22,
    "position": 8,
    "team": "RB Honda RBPT",
    "time": "1:16.788",
  },
  {
    "code": "STR",
    "name": "Lance Stroll",
    "number": 18,
    "position": 9,
    "team": "Aston Martin Aramco Mercedes",
    "time": "1:17.072",
  },
  {
    "code": "ALO",
    "name": "Fernando Alonso",
    "number": 14,
    "position": 10,
    "team": "Aston Martin Aramco Mercedes",
    "time": "1:17.552",
  },
  {
    "code": "HAM",
    "name": "Lewis Hamilton",
    "number": 44,
    "position": 11,
    "team": "Mercedes",
    "time": "1:16.960",
  },
  {
    "code": "ALB",
    "name": "Alexander Albon",
    "number": 23,
    "position": 12,
    "team": "Williams Mercedes",
    "time": "1:17.167",
  },
  {
    "code": "BOT",
    "name": "Valtteri Bottas",
    "number": 77,
    "position": 13,
    "team": "Kick Sauber Ferrari",
    "time": "1:17.340",
  },
  {
    "code": "MAG",
    "name": "Kevin Magnussen",
    "number": 20,
    "position": 14,
    "team": "Haas Ferrari",
    "time": "1:17.427",
  },
  {
    "code": "OCO",
    "name": "Esteban Ocon",
    "number": 31,
    "position": 15,
    "team": "Alpine Renault",
    "time": "1:17.697",
  },
  {
    "code": "HUL",
    "name": "Nico Hulkenberg",
    "number": 27,
    "position": 16,
    "team": "Haas Ferrari",
    "time": "1:17.976",
  },
  {
    "code": "GAS",
    "name": "Pierre Gasly",
    "number": 10,
    "position": 17,
    "team": "Alpine Renault",
    "time": "1:17.982",
  },
  {
    "code": "RIC",
    "name": "Daniel Ricciardo",
    "number": 3,
    "position": 18,
    "team": "RB Honda RBPT",
    "time": "1:18.085",
  },
  {
    "code": "ZHO",
    "name": "Zhou Guanyu",
    "number": 24,
    "position": 19,
    "team": "Kick Sauber Ferrari",
    "time": "1:18.188",
  },
]
`;
